causallearn/utils/cit.py

```diff
@@ -59,9 +59,18 @@ def _unique(column):
         }
 
     def kci(self, X, Y, condition_set):
+        if condition_set == None:
+            condition_set = []
+        if type(X) == int:
+            X = [X]
+        if type(Y) == int:
+            Y = [Y]
+        if type(condition_set) == int:
+            condition_set = [condition_set]
+        
         if len(condition_set) == 0:
-            return self.kci_ui.compute_pvalue(self.data[:, [X]], self.data[:, [Y]])[0]
-        return self.kci_ci.compute_pvalue(self.data[:, [X]], self.data[:, [Y]], self.data[:, list(condition_set)])[0]
+            return self.kci_ui.compute_pvalue(self.data[:, X], self.data[:, Y])[0]
```

Comment:
great, thanks!

QQ:
1. what's the reason that the old code only support one variable? Is there some special design? @MarkDana

Reply:
@tofuwen There is no special design in the old code for supporting only one variable. I just aligned with other tests (used in constraint-based methods), and forgot that KCI can take in multivariate unconditional variables.

I was just about to fix it. So thanks so much for your work @cogito233 !

Reply:
Oh but the citest cache should also be fixed accordingly (see https://github.com/cmu-phil/causal-learn/blob/ffe75f95c4003fa7e9d7d5f3bbec4ace90ed3a41/causallearn/utils/cit.py#L339). I'll handle this.

Reply:
For https://github.com/cmu-phil/causal-learn/blob/ffe75f95c4003fa7e9d7d5f3bbec4ace90ed3a41/causallearn/utils/cit.py#L338,

we'll need to handle the case for iterable `X`, `Y`.

Reply:
Also for the cache key in https://github.com/cmu-phil/causal-learn/blob/ffe75f95c4003fa7e9d7d5f3bbec4ace90ed3a41/causallearn/utils/cit.py#L339,

we'll need to handle `X < Y` for iterable `X`, `Y`. And also `frozenset(i)` as hashable key.

Reply:
> For
> 
> https://github.com/cmu-phil/causal-learn/blob/ffe75f95c4003fa7e9d7d5f3bbec4ace90ed3a41/causallearn/utils/cit.py#L338
> 
> ,
> we'll need to handle the case for iterable `X`, `Y`.

I have no ideas about how to change the assert in the case of `kernel CIT`(Concerning the whole or part of `X` in the `conditional set`). Other problems are already fixed. 

Maybe $X \bot Y | X$ is also a valid expression?

Reply:
Wow cool. You're so productive! Thanks for all these!

I see your point. This line was intended for correctness checks in constraint-based methods.

As for the citest itself, is `X;Y|X` valid? I don't know actually - but the results is expected to be always "independent" (consider `X|X` as degenerated const).

How about we force `X not in condition_set` for `X: int`, and `len(set(condition_set).intersection(X)) == 0` for `X: Iterable`?